account_tax_balance/models/account_tax.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# © 2016 Lorenzo Battistini - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields
+
+
+class AccountTax(models.Model):
+    _inherit = 'account.tax'
+
+    balance = fields.Float(string="Balance", compute="_compute_balance")
+    base_balance = fields.Float(
+        string="Base Balance", compute="_compute_balance")
+
+    def get_context_values(self):
+        if not self.env.context.get('from_date'):
+            from_date = fields.Date.context_today(self)
+        else:
+            from_date = self.env.context['from_date']
+        if not self.env.context.get('to_date'):
+            to_date = fields.Date.context_today(self)
+        else:
+            to_date = self.env.context['to_date']
+        if not self.env.context.get('target_move'):
+            target_move = 'posted'
+        else:
+            target_move = self.env.context['target_move']
+        if not self.env.context.get('company_id'):
+            company_id = self.env.user.company_id.id
+        else:
+            company_id = self.env.context['company_id']
+        return from_date, to_date, company_id, target_move
+
+    def _compute_balance(self):
+        from_date, to_date, company_id, target_move = self.get_context_values()
+        for tax in self:
```

Comment:
By splitting the two methods `compute_balance` and `compute_base_balance` you introduced duplication. If you did everything together you could just write the `balance` and `base_balance` in-place in the same method instead of returning the value and writing it here. What do you think?